src/plugins/length/length.c

```diff
@@ -0,0 +1,104 @@
+/**
+ * @file
+ *
+ * @brief Source for length plugin
+ *
+ * @copyright BSD License (see doc/LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+#include <kdberrors.h>
+#include <regex.h>
+#include <stdio.h>
+
+#include "length.h"
+
+
+static int validateKey (Key * key, Key * parentKey)
+{
+	const Key * meta = keyGetMeta (key, "check/length");
+	if (!meta) return 1;
+	int rc = 0;
+	int c = 0;
+	const char * length = keyString (meta);
+	int i;
+	sscanf (length, "%d", &i);
+
+	const char * text = keyString (key);
+
+	while (text[c] != '\0')
+	{
+		c++;
+	}
+	if (c <= i)
+	{
+		rc = true;
+	}
+	else
+	{
+		rc = false;
+	}
+
+
+	if (!rc)
+	{
+		ELEKTRA_SET_VALIDATION_SEMANTIC_ERRORF (parentKey, "Validation of key %s with value %s failed", keyName (key),
```

Comment:
Please improve the error message: write that the length check failed and with which length.